nixos/modules/services/networking/unbound.nix

```diff
@@ -84,31 +161,82 @@ in
         description = "Use and update root trust anchor for DNSSEC validation.";
       };
 
-      extraConfig = mkOption {
-        default = "";
-        type = types.lines;
-        description = ''
-          Extra unbound config. See
-          <citerefentry><refentrytitle>unbound.conf</refentrytitle><manvolnum>8
-          </manvolnum></citerefentry>.
+      settings = mkOption {
+        type = with types; submodule settingsSubmodule;
```

Comment:
I'm curious why you went to the trouble of defining the top level attributes (`server`, `forward-zone`, etc...) when you could have used a more generic approach. Assuming these top level attributes are stable (new ones aren't often introduced) I think you get bonus points for being thorough :+1:

Reply:
Those top-level attributes indeed are stable and have different types. That's why I chose to explicitly define them.

Reply:
> could have used a more generic approach

@aanderse As stated in the RFC it's perfectly fine (and IMHO even mandatory) to keep those for common options (and I think that `server`/`forward-zone`/etc can be considered common).

Reply:
Yeah was just curious. I think this is great.